bundles/org.openhab.binding.sony/src/main/java/org/openhab/binding/sony/internal/providers/SonyDefinitionProviderImpl.java

```diff
@@ -0,0 +1,384 @@
+                    return groupType;
+                }
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public Collection<ChannelGroupType> getChannelGroupTypes(final @Nullable Locale locale) {
+        final Map<ChannelGroupTypeUID, ChannelGroupType> groupTypes = new HashMap<>();
+        for (final SonySource src : sources) {
+            final Collection<ChannelGroupType> localGroupTypes = src.getChannelGroupTypes();
+            if (localGroupTypes != null) {
+                for (final ChannelGroupType gt : localGroupTypes) {
+                    if (!groupTypes.containsKey(gt.getUID())) {
+                        groupTypes.put(gt.getUID(), gt);
+                    }
+                }
+            }
+        }
+        return groupTypes.values();
+    }
+
+    @Override
+    public Collection<ThingType> getThingTypes(final @Nullable Locale locale) {
+        final Map<ThingTypeUID, ThingType> thingTypes = new HashMap<>();
+        for (final SonySource src : sources) {
+            for (final ThingType tt : src.getThingTypes()) {
+                if (!thingTypes.containsKey(tt.getUID())) {
+                    thingTypes.put(tt.getUID(), tt);
+                }
+            }
+        }
+        return thingTypes.values();
+    }
+
+    @Override
+    public @Nullable ThingType getThingType(final ThingTypeUID thingTypeUID, final @Nullable Locale locale) {
+        Objects.requireNonNull(thingTypeUID, "thingTypeUID cannot be null");
+        if (StringUtils.equalsIgnoreCase(thingTypeUID.getBindingId(), SonyBindingConstants.BINDING_ID)) {
+            for (final SonySource src : sources) {
+                final ThingType thingType = src.getThingType(thingTypeUID);
+                if (thingType != null) {
+                    return thingType;
+                }
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public void addStateOverride(final ThingUID thingUID, final String channelId,
+            final StateDescription stateDescription) {
+        Objects.requireNonNull(thingUID, "thingUID cannot be null");
+        Validate.notEmpty(channelId, "channelId cannot be empty");
+        Objects.requireNonNull(stateDescription, "stateDescription cannot be null");
+
+        final ChannelUID id = new ChannelUID(thingUID, channelId);
+        stateOverride.put(id, stateDescription);
+    }
+
+    @Override
+    public @Nullable StateDescription getStateDescription(final Channel channel,
+            final @Nullable StateDescription originalStateDescription, final @Nullable Locale locale) {
+        Objects.requireNonNull(channel, "channel cannot be null");
+
+        if (StringUtils.equalsIgnoreCase(channel.getUID().getBindingId(), SonyBindingConstants.BINDING_ID)) {
+            return getStateDescription(channel.getUID().getThingUID(), channel.getUID().getId(),
+                    originalStateDescription);
+        }
+        return originalStateDescription;
```

Comment:
Please return `null` if no description will be provided for the channel.

See

https://github.com/openhab/openhab-core/blob/3811cb95604f69f6a3d77da2caf391d81b05458e/bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/type/DynamicStateDescriptionProvider.java#L30-L43.